Update Travis CI configuration

gwils opened the pull request:
Add GHC 8.4.3
Add GHC 8.6.1 alpha/beta as an allowed failure

conal commented:
Thanks for the PR. There's a lot going on here that is opaque to me. (I know almost nothing about Travis.) Would you please describe these changes?

gwils commented:
There are three classes of change here:

- addition of GHC 8.4.3 as a required pass
- addition of GHC 8.6 alpha as an allowed failure
- changes to Herbert's script that have occurred since last time the Travis file was generated

In order:

Testing GHC 8.4 is useful since by now many people are using this release and I assume you'd like to support it going forward.

The large change from line 81 to around 100 sets up the necessary conditions for GHC 8.6 alpha to be tested. This build is an "allowed failure", so the Travis build will still pass even if 8.6 fails. With the release of 8.6 around the corner, this can be a useful thing to look at. If you aren't interested though, I can remove this part.

This Travis file is generated by [Herbert's script](https://github.com/haskell-CI/haskell-ci) which works by reading the `tested-with` field of the cabal file and generating an appropriate `.travis.yml` file. I find using this script much easier than writing a travis configuration by hand, so I use it for all my projects. i find it easier to just re-run the script than to modify its output manually.
When the script changes, the output .travis.yml files can change, as we've experienced here. 

So there we are. If that doesn't sound good, would you prefer if I split this out into three small commits so that it's clear which lines are related each of these? Would you prefer if I removed the GHC 8.6 alpha testing?

conal commented:
Thanks very much for the PR and the explanation. I hadn't heard of Herbert's script.

[Released to Hackage](http://hackage.haskell.org/package/checkers-0.4.11).
